packages/quantic/force-app/main/default/lwc/quanticSort/quanticSort.js

```diff
@@ -148,6 +236,37 @@ export default class QuanticSort extends LightningElement {
   }
 
   get value() {
-    return this.state?.sortCriteria;
+    return this.hasCustomSortOptions
+      ? this.customSortOptions[0].value
+      : this.state?.sortCriteria;
+  }
+
+  get hasCustomSortOptions() {
+    return this.customSortOptions.length > 0;
+  }
+
+  get hasError() {
+    return this.hasInitializationError || !!this.errorMessage;
+  }
+
+  /**
+   * Returns an array of custom sort options passed via slots.
+   * @returns {SortOption[]} The specified custom sort options.
+   */
+  get customSortOptions() {
+    /** @type {SortOption[]} */
+    // @ts-ignore
+    const elements = Array.from(this.querySelectorAll('c-quantic-sort-option'));
+    if (elements.length === 0) {
+      return [];
+    }
```

Comment:
```suggestion
```
By using `Array.from` you ensure it's an array.

And calling `.map` on an empty array will simply do nothing and return an empty array.

Reply:
yeah makes sense